### PR TITLE
add action to build conda environment upon changes

### PR DESCRIPTION
**Pull request by phue**
Many thanks to contributing to nf-core/methylseq!

Please fill in the appropriate checklist below (delete whatever is not relevant). These are the most common things requested on pull requests (PRs).

## PR checklist
 - [ ] This comment contains a description of changes (with reason)
 - [ ] If you've fixed a bug or added code that should be tested, add tests!
 - [ ] If necessary, also make a PR on the [nf-core/methylseq branch on the nf-core/test-datasets repo]( https://github.com/nf-core/test-datasets/pull/new/nf-core/methylseq)
 - [ ] Ensure the test suite passes (`nextflow run . -profile test,docker`).
 - [ ] Make sure your code lints (`nf-core lint .`).
 - [ ] Documentation in `docs` is updated
 - [ ] `CHANGELOG.md` is updated
 - [ ] `README.md` is updated

**Learn more about contributing:** https://github.com/nf-core/methylseq/tree/master/.github/CONTRIBUTING.md

This is more of a proposal:

Currently, the docker build is only triggered _after_ a merge commit, which has caused problems in the past when the build failed (particularly when bumping versions of conda packages).

This action triggers a test build of the updated conda environment, _if_ `environment.yml` has changed.

Assuming that:

* docker build failures are likely caused by conda version conflicts
* all nf-core pipelines follow this mechanism of baking a conda env into a docker container

.. I believe this could improve our chances of catching such problems earlier, before a broken container causes further problems. So I propose adding such an action for all pipelines via an upcoming template sync

### CODE REVIEW

File: .github/workflows/conda.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    paths:
+      - 'environment.yml'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    
+    steps:
+      - uses: actions/checkout@v2
+      - uses: docker://continuumio/miniconda3:4.8.2
```
Comment (ewels) [inline .github/workflows/conda.yml:15]:
```suggestion
      - uses: docker://nfcore/base:latest
```

**Comment by ewels** [comment]
So yeah, more complete feedback is that I would like to have something like this for the future:

* CI workflow has two additional steps:
    * One to see if `environment.yaml` has changed (action needs to always run, so can't use `path`
    * One to build and tag a new container - skipped if the above is false
    * If no change, then pull the container as currently (skip if we're building)
* Run tests with either the pulled container or the new, updated container
* Final step, so only if all others passed, push to dockerhub if we built a new container

nf-core/tools#595 has some extra complication in the discussion revolving around how we handle tagging released containers. But I think that we all agreed that it's basically ok. So just needs writing 😁

**Comment by phue** [comment]
Alright, it seem I should to read up on current open issues to get in the loop again (been busy with other things for the last couple of months)

So I suppose we leave this as a draft pending a final solution for https://github.com/nf-core/tools/issues/595

**Comment by ewels** [comment]
No worries, it was a recent discussion and there are hundreds of issues across @nf-core these days 😄 

I'm happy to merge this and delete it again once the upstream GHA docker-build stuff is written. This is nice and clean and will be a helpful tool until then, and you've already gone to the effort of writing it 😅

File: .github/workflows/conda.yml (hunk 1 above)
Comment (ewels) [reply .github/workflows/conda.yml:15]: `nfcore/base` is essentially the same container, and is the one we'll be using for the main build. So may as well use it here as well I figure..

**Comment by phue** [comment]
Sounds good :+1:  

I made the switch to the nf-core container as you suggested.

But the irreproducible error is back. My feeling is that it has to do ith the way github actions runs the build matrix.
The failing process relies on a previous run that has the `--save_reference` parameter set, so that `results/reference_genome/genome.fa.*` exists on the build runner.
It could be that the order of a build matrix is not garantueed, but still I would expect nextflow to throw an error before bwameth does, because we have `checkIfExists` set:
https://github.com/nf-core/methylseq/blob/4f31ed1792e3cc8ce0ea648b6ee8f541f70102f7/main.nf#L147

Unfortunately it is hard to debug the error, I somehow miss the Travis feature of ssh'ing into the build host.

**Comment by ewels** [comment]
Super weird. I thought about the matrix too, but all three commands are in the same step so it should be exactly the same environment, just three commands run one after another..

**Pull request merged**